Sources/Networking/NetworkClient/NetworkClient.swift

```diff
@@ -104,4 +104,9 @@ public extension NetworkClient {
   func request(_ urlRequest: URLRequest) -> AnyPublisher<Response, NetworkError> {
     performRequest(urlRequest)
   }
+
+  func request(_ urlRequest: URLRequest) async throws -> Response {
```

Comment:
I dont think this is necessary here, or in case I am wrong, what is the purpose? The `NetworkClient` should be accessed via the `NetworkClientType` only. `NetworkClientType` already provides its synthetized `async` functions... Same applies to the `AuthorizedNetworkClient`.